sdk/python/tests/test_kfp_samples.sh

```diff
@@ -70,14 +71,22 @@ cp "${COMPILE_REPORT_FILE}" "${COMPILE_REPORT_FILE_OLD}"
 # delete the previous compiler output file
 rm -f "${COMPILER_OUTPUTS_FILE}"
 
+# check which pipelines have special configurations
+PIPELINES=$(awk '/pipeline:/{print $NF}' ${CONFIG_FILE})
+
 # compile each of the Python scripts in the KFP testdata folder
 for f in "${KFP_TESTDATA_DIR}"/*.py; do
   echo -e "\nCompiling ${f##*/}:" >> "${COMPILER_OUTPUTS_FILE}"
-  if dsl-compile-tekton --py "${f}" --output "${TEKTON_COMPILED_YAML_DIR}/${f##*/}.yaml" >> "${COMPILER_OUTPUTS_FILE}" 2>&1;
-  then
-    echo "SUCCESS: ${f##*/}" | tee -a "${COMPILER_OUTPUTS_FILE}"
+  IS_SPECIAL=$(grep -E ${f##*/} <<< ${PIPELINES})
```

Comment:
For the `test_kfp_samples.sh` script, why do we not just generate `PipelineRun` for all `testdata` scripts and only change one line:

```Bash
  if dsl-compile-tekton --py "${f}" --output "${TEKTON_COMPILED_YAML_DIR}/${f##*/}.yaml" >> "${COMPILER_OUTPUTS_FILE}" 2>&1;
```
to
```Bash
  if dsl-compile-tekton --generate-pipelinerun --py "${f}" --output "${TEKTON_COMPILED_YAML_DIR}/${f##*/}.yaml" >> "${COMPILER_OUTPUTS_FILE}" 2>&1;
```
There is no need or benefit to have a mix here, since we will catch the `NotImplementedError`s either way.

Reply:
Maybe I do not understand, but wouldn't this still fail to compile basic_no_decorator.py and compose.py?

Reply:
Hi Andrew, yes, I wrote that comment before understanding the full purpose of your `test_util.py`. But let me explain the my bigger point below